Automattic/newspack-plugin
# feat: add multi-select capabillity to AutocompleteWithSuggestions

### All Submissions:






### Changes proposed in this Pull Request:



This complete the `AutocompleteWithSuggestions` component to add multi-select capability. This is needed for the proposed child listings UI design in Newspack Listings.

Closes #954.

### How to test the changes in this Pull Request:

1. Navigate to the components demo page. Confirm there are now two instances of the `AutocompleteWithSuggestions` component: the first is single-select (unchanged from `master`), the second is multi-select with checkboxes instead of links.
2. Test the single-select UI; its behavior should be unchanged from `master`, except that the selected item is now shown at the top of the component separate from the autocomplete field, which is now only used for search functionality:

<img width="833" alt="Screen Shot 2021-05-21 at 11 52 54 AM" src="https://user-images.githubusercontent.com/2230142/119178524-172b3600-ba2b-11eb-86ef-374bb0f15c99.png">

3. Test the multi-select UI:
  - Post type selector: This dropdown lets you change which post type is shown in suggestions, both in the autocomplete field and in the suggestions list. (This can actually be used in both single-select and multi-select flavors of the component, but is only shown in multi-select to demonstrate that it's optional.)
  - Multi-select capability: You should now be able to select more than one item. Selected items appear at the top of the component, and are checked if displayed in the suggestions list. Checking an item from the list should add it to the selections, unchecking it should remove it. Typing a post title into the autocomplete field and selecting a suggestion should add the suggestion if it's not already selected, and remove it if it is.
  - "Clear all" feature: If more than one item is selected, a "Clear all" button will appear at the top of the component. Clicking this will remove all selections and return the component to an empty state.

<img width="833" alt="Screen Shot 2021-05-21 at 12 00 08 PM" src="https://user-images.githubusercontent.com/2230142/119179423-1d6de200-ba2c-11eb-96b7-fb1fe325e117.png">

4. Test load more functionality: For both single-select and multi-select flavors, if there are more posts to suggest than the `suggestionsToFetch` prop (default: 10), a "Load more" button will appear at the end of the suggestions list. Clicking this button will load `suggestionsToFetch` more  posts until all posts of the selected post type are loaded.

### Other information:
